### ISSUE TITLE
Regression with 1.24.0: ineffassign: wrong type for method

### ISSUE THREAD

**Issue by agnivade** [golangci/golangci-lint#1014]
Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
1.24.0
```

</details>

<details><summary>Config file</summary>
https://github.com/mattermost/mattermost-server/blob/master/.golangci.yml

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
1.14.1
```

</details>

<details><summary>Verbose output of running</summary>
https://build.mattermost.com/blue/organizations/jenkins/mp%2Fmattermost-server-pr-new/detail/bumpgolangci/6/pipeline

</details>

Especially, I have a long line of errors like this: 

```
level=error msg="Running error: ineffassign: analysis skipped: errors in package: [/go/src/github.com/mattermost/mattermost-server/plugin/helpers_bots_test.go:36:11: cannot use api (variable of type *plugintest.API) as plugin.API value in assignment: wrong type for method AddChannelMember /go/src/github.com/mattermost/mattermost-server/plugin/helpers_bots_test.go:50:12: cannot use api (variable of type *plugintest.API) as plugin.API value in assignment: wrong type for method AddChannelMember /go/src/github.com/mattermost/mattermost-server/plugin/helpers_bots_test.go:60:12: cannot use api (variable of type *plugintest.API) as plugin.API value in assignment: wrong type for method AddChannelMember /go/src/github.com/mattermost/mattermost-server/plugin/helpers_bots_test.go:79:12: cannot use api (variable of type *plugintest.API) as plugin.API value in assignment: wrong type for method AddChannelMember /go/src/github.com/mattermost/mattermost-server/plugin/helpers_bots_test.go:94:12: cannot use api (variable of type *plugintest.API) as plugin.API value in assignment: wrong type for method 
```

**Comment by hairyhenderson** [comment]
I believe I'm also seeing the same issue:

```
WARN [runner] Can't run linter goanalysis_metalinter: S1038: failed prerequisites: [(inspect@github.com/hairyhenderson/gomplate/v3/tests/integration [github.com/hairyhenderson/gomplate/v3/tests/integration.test], isgenerated@github.com/hairyhenderson/gomplate/v3/tests/integration [github.com/hairyhenderson/gomplate/v3/tests/integration.test]): analysis skipped: errors in package: [/go/src/github.com/hairyhenderson/gomplate/tests/integration/test_ec2_utils.go:85:33: cannot use cert (variable of type *x509.Certificate) as *x509.Certificate value in argument to signedData.AddSigner]] 
ERRO Running error: S1038: failed prerequisites: [(inspect@github.com/hairyhenderson/gomplate/v3/tests/integration [github.com/hairyhenderson/gomplate/v3/tests/integration.test], isgenerated@github.com/hairyhenderson/gomplate/v3/tests/integration [github.com/hairyhenderson/gomplate/v3/tests/integration.test]): analysis skipped: errors in package: [/go/src/github.com/hairyhenderson/gomplate/tests/integration/test_ec2_utils.go:85:33: cannot use cert (variable of type *x509.Certificate) as *x509.Certificate value in argument to signedData.AddSigner]] 
```

(from https://app.circleci.com/pipelines/github/hairyhenderson/gomplate/1272/workflows/ec300649-d694-48c4-a19e-20edb3071c29/jobs/5568)

What's strange is that it seems inconsistent - I've never been able to reproduce locally, and if I rebuild the CI job it sometimes works fine.

**Comment by ernado** [comment]
Please try v1.25

**Comment by ldez** [comment]
Fixed by v1.25
